src/clj/puppetlabs/services/request_handler/request_handler_core.clj

```diff
@@ -325,6 +325,20 @@
       (.handleRequest (:jruby-instance request))
       response->map)))
 
+(defn get-environment-from-request
+  "Gets the environment from a request, after the params map has been populated
+  via wrap-params-for-jruby."
+  [req]
+  ((:params (wrap-params-for-jruby req)) "environment"))
+
+(defn catalog-request-handler
+  "Specialized handler for catalogs. Adds code_id to the request before passing
```

Comment:
The usual naming convention for middleware is `wrap-*` or `wrap-with-*`.   So I'd actually go `wrap-with-code-id` or something like that.